datatypes/src/operations/image/colorizer.rs

```diff
@@ -3,44 +3,140 @@ use crate::operations::image::RgbaTransmutable;
 use crate::raster::Pixel;
 use crate::util::Result;
 use ordered_float::{FloatIsNan, NotNan};
-use serde::{Deserialize, Serialize};
+use serde::de::{MapAccess, Visitor};
+use serde::{Deserialize, Deserializer, Serialize};
 use snafu::ensure;
 use std::collections::HashMap;
 use std::convert::TryFrom;
+use std::fmt::Formatter;
 use std::str::FromStr;
 
 /// A colorizer specifies a mapping between raster values and an output image
 /// There are different variants that perform different kinds of mapping.
-#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Serialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase", tag = "type")]
 pub enum Colorizer {
     #[serde(rename_all = "camelCase")]
     LinearGradient {
         breakpoints: Breakpoints,
         no_data_color: RgbaColor,
-        default_color: RgbaColor,
+        over_color: RgbaColor,
+        under_color: RgbaColor,
     },
     #[serde(rename_all = "camelCase")]
     LogarithmicGradient {
         breakpoints: Breakpoints,
         no_data_color: RgbaColor,
-        default_color: RgbaColor,
+        over_color: RgbaColor,
+        under_color: RgbaColor,
     },
     #[serde(rename_all = "camelCase")]
     Palette {
         colors: Palette,
         no_data_color: RgbaColor,
-        default_color: RgbaColor,
+        over_color: RgbaColor,
+        under_color: RgbaColor,
```

Comment:
For palettes, it does not make sense to have an over or under color, since this means exact values.
We should keep the default color here.

Reply:
Palette now uses default color again. https://github.com/geo-engine/geoengine/pull/684/commits/f0a0bd67359b995ca311acbb57ac04b7707288b8